pkgs/top-level/impure.nix

```diff
@@ -65,13 +74,9 @@ in
           Please remove one of them and try again.
         ''
       else if builtins.pathExists homeOverlaysFile then
-        if isDir homeOverlaysFile then
-          throw (homeOverlaysFile + " should be a file")
-        else overlays homeOverlaysFile
+        overlaysFile homeOverlaysFile
       else if builtins.pathExists homeOverlaysDir then
-        if !(isDir homeOverlaysDir) then
-          throw (homeOverlaysDir + " should be a directory")
-        else overlays homeOverlaysDir
+        overlaysDir homeOverlaysDir
```

Comment:
This leads to a worse error message, the `isDir` check should be kept because of that.